### ISSUE TITLE
Improve error recovery when parsing an incomplete foreach loop 

### ISSUE THREAD

**Issue by jcouv** [dotnet/roslyn#27633]
Before deconstruction was introduced (in C# 7.0), an incomplete foreach loop like `foreach (var in collection) ...` would parse as a `ForEachStatementSyntax`.
But as of dev15, it started parsing as a deconstruction foreach (`ForEachVariableStatementSyntax`). This is unexpected.

Relates to https://github.com/dotnet/roslyn/pull/27605

**Comment by bkoelman** [comment]
Yes, I ran into that too. Probably an unintentional, but breaking change.

**Comment by CyrusNajmabadi** [comment]
Going to close this out.  I don't think we need to do anything here as we haven't heard any reports about this being a problem
